File: solutions/ruby/acronym/1/acronym.rb
module Acronym
  def self.abbreviate(string)
    string.split(/[\s\-_]+/)
          .reject(&:empty?)
          .map { |word| word.gsub(/[^a-zA-Z]/, '')[0]&.upcase }
          .compact
          .join
  end
end

File: solutions/elixir/acronym/1/lib/acronym.ex
defmodule Acronym do
  @doc """
  Generate an acronym from a string.
  "This is a string" => "TIAS"
  """
  @spec abbreviate(String.t()) :: String.t()
  def abbreviate(string) do
    String.split(string, ~r/\s|-|_/)
    |> Enum.map(&String.upcase/1)
    |> Enum.map(fn word -> String.first(word) end)
    |> Enum.join("")
  end
end
